lib/App/KSP_CKAN/Tools/NetKAN.pm

```
@@ -0,0 +1,119 @@
+
+  use App::KSP_CKAN::Tools::NetKAN;
+
+  my $netkan = App::KSP_CKAN::Tools::NetKAN->new(
+    netkan => "/path/to/netkan.exe",
+    chache => "/path/to/cache",
+    token => $token,
+    file => '/path/to/file.netkan',
+    ckan_meta => '/path/to/CKAN-meta',
+  );
+
+=head1 DESCRIPTION
+
+Is a wrapper for the NetKAN inflater. Initially it will
+just wrap and capture errors, but the intention is to 
+add helper methods to check for changes in remote meta
+data and only run the inflater when required.
+
+=cut
+
+my $Ref = sub {
+  croak("auth isn't a 'App::KSP_CKAN::Tools::Config' object!") unless reftype( $_[0] )->class eq "App::KSP_CKAN::Tools::Config";
+};
+
+has 'config'      => ( is => 'ro', required => 1, isa => $Ref );
+has 'netkan'    => ( is => 'ro', required => 1 );
+has 'cache'     => ( is => 'ro', default => sub { File::Spec->tmpdir()."/NetKAN-cache"; } );
+has 'file'      => ( is => 'ro', required => 1);
+has 'ckan_meta' => ( is => 'ro', required => 1 );
+has 'token'     => ( is => 'ro' );
+has 'rescan'    => ( is => 'ro', default => sub { 1 } );
+has '_output'   => ( is => 'ro', lazy => 1, builder => 1 );
+has '_cli'      => ( is => 'ro', lazy => 1, builder => 1 );
+has '_cache'    => ( is => 'ro', lazy => 1, builder => 1 );
+
+method _build__cache {
+  if ( ! -d $self->cache ) {
+    mkpath($self->cache);
+  }
+
+  return $self->cache;
+}
+
+method _build__output {
+  my $basename = basename($self->file,  ".netkan");
+  if (! -d $self->ckan_meta."/".$basename ) {
+    mkdir $self->ckan_meta."/".$basename;
+  }
+  return $self->ckan_meta."/".$basename;
+}
+
+method _build__cli {
+  if ($self->token) {
+    return $self->netkan." --outputdir=".$self->_output." --cachedir=".$self->_cache." --github-token=".$self->token." ".$self->file;
+  } else {
+    return $self->netkan." --outputdir=".$self->_output." --cachedir=".$self->_cache." ".$self->file;
+  }
+}
+
+method _check_lite {
+  # TODO: Build a method to go and check if required full inflate
```

To be safe I'd be inclined to throw an exception here rather than just return zero. (Unless it's actually acceptable for this method to do nothing.)